client/driver/rkt.go

```diff
@@ -0,0 +1,203 @@
+package driver
+
+import (
+        "bytes"
+        "encoding/json"
+        "fmt"
+        "log"
+        "os"
+        "os/exec"
+        "regexp"
+        "runtime"
+        "strings"
+        "syscall"
+        "time"
+
+        "github.com/hashicorp/nomad/client/config"
+        "github.com/hashicorp/nomad/nomad/structs"
+)
+
+var (
+        reRktVersion  = regexp.MustCompile("rkt version ([\\d\\.]+).+")
+        reAppcVersion = regexp.MustCompile("appc version ([\\d\\.]+).+")
+)
+
+// RktDriver is a driver for running images via Rkt
+// We attempt to chose sane defaults for now, with more configuration available
+// planned in the future
+type RktDriver struct {
+        DriverContext
+}
+
+// rktHandle is returned from Start/Open as a handle to the PID
+type rktHandle struct {
+        proc   *os.Process
+        name   string
+        waitCh chan error
+        doneCh chan struct{}
+}
+
+// rktPID is a struct to map the pid running the process to the vm image on
+// disk
+type rktPID struct {
+        Pid  int
+        Name string
+}
+
+// NewRktDriver is used to create a new exec driver
+func NewRktDriver(ctx *DriverContext) Driver {
+        return &RktDriver{*ctx}
+}
+
+func (d *RktDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
+        // Only enable if we are root when running on non-windows systems.
+        if runtime.GOOS != "windows" && syscall.Geteuid() != 0 {
+                d.logger.Printf("[DEBUG] driver.rkt: must run as root user, disabling")
+                return false, nil
+        }
+
+        outBytes, err := exec.Command("rkt", "version").Output()
+        if err != nil {
+                return false, nil
+        }
+        out := strings.TrimSpace(string(outBytes))
+
+        rktMatches := reRktVersion.FindStringSubmatch(out)
+        appcMatches := reRktVersion.FindStringSubmatch(out)
+        if len(rktMatches) != 2 || len(appcMatches) != 2 {
+                return false, fmt.Errorf("Unable to parse Rkt version string: %#v", rktMatches)
+        }
+
+        node.Attributes["driver.rkt"] = "1"
+        node.Attributes["driver.rkt.version"] = rktMatches[0]
+        node.Attributes["driver.appc.version"] = appcMatches[1]
+
+        return true, nil
+}
+
+// Run an existing Rkt image.
+func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
+        trust_prefix, ok := task.Config["trust_prefix"]
+        if !ok || trust_prefix == "" {
+                return nil, fmt.Errorf("Missing trust prefix for rkt")
+        }
+        trust_args := []string{
+                "sudo",
```

Comment:
Since we check in the fingerprint that we are running as root, I don't think we will need `sudo` here.